### ISSUE TITLE
TemplateStyles: Include templatestyles in html.head and dedupe

### ISSUE THREAD

**Issue by gnosygnu** [gnosygnu/xowa#314]
&lt;templatestyles> has been changed to do the following:
* Place &lt;style> in html.head
* Dedupe &lt;style> so it's not repeated

This implementation differs from MediaWiki because of complexities due to tidy and inline styles.

**Comment by desb42** [comment]
I have just integrated your code changes and I now get
With tidy
<img width="859" alt="tempstyle_withtidy" src="https://user-images.githubusercontent.com/861631/50570803-209ba100-0d91-11e9-867d-557ec49f6c5c.png">

and without
![tempstyle_withouttidy](https://user-images.githubusercontent.com/861631/50570805-2db89000-0d91-11e9-890b-b92d5884eec8.png)

it has helped the non tidy version best

**Comment by gnosygnu** [comment]
Yeah, there's another issue wherein it needs to download the ext.cite.styles . I think that should help the tidy issue, but haven't gotten around to updating the cite code
